test/controllers/check.your.answers.controller.spec.ts

```diff
@@ -731,6 +731,92 @@ describe("GET tests", () => {
     expect(stringCount(BO_NOC_HEADING, resp.text)).toEqual(3);
     expect(stringCount(TRUSTS_NOC_HEADING, resp.text)).toEqual(2); // BO Gov doesn't have the trustee of a trust NOC
     expect(stringCount(FIRM_NOC_HEADING, resp.text)).toEqual(3);
+    expect(stringCount(FIRM_CONTROL_NOC_HEADING, resp.text)).toEqual(0);
+    expect(stringCount(TRUST_CONTROL_NOC_HEADING, resp.text)).toEqual(0);
+    expect(stringCount(OWNER_OF_LAND_PERSON_NOC_HEADING, resp.text)).toEqual(0);
+    expect(stringCount(OWNER_OF_LAND_OTHER_ENITY_NOC_HEADING, resp.text)).toEqual(0);
+
+    // Count the NOCs
+    expect(stringCount(BO_NOC_OVER_25_PERCENT_OF_SHARES, resp.text)).toEqual(3);
+    expect(stringCount(BO_NOC_OVER_25_PERCENT_OF_VOTING_RIGHTS, resp.text)).toEqual(3);
+    expect(stringCount(BO_NOC_APPOINT_OR_REMOVE_MAJORITY_BOARD_DIRECTORS, resp.text)).toEqual(3);
+    expect(stringCount(BO_NOC_SIGNIFICANT_INFLUENCE_OR_CONTROL, resp.text)).toEqual(3);
+
+    // The same NOC text is used for 'trustee of a trust' and 'control of trust' Nocs.
+    // The BO Gov only has 'control of trust' and not 'trustee of a trust' so these NOCs should appear 5 times
+    expect(stringCount(escapeNOCText(BO_NOC_TRUSTEE_OF_TRUST_OVER_25_PERCENT_OF_SHARES), resp.text)).toEqual(2);
+    expect(stringCount(escapeNOCText(BO_NOC_TRUSTEE_OF_TRUST_OVER_25_PERCENT_OF_VOTING_RIGHTS), resp.text)).toEqual(2);
+    expect(stringCount(escapeNOCText(BO_NOC_TRUSTEE_OF_TRUST_APPOINT_OR_REMOVE_MAJORITY_BOARD_DIRECTORS), resp.text)).toEqual(2);
+    expect(stringCount(escapeNOCText(BO_NOC_TRUSTEE_OF_TRUST_SIGNIFICANT_INFLUENCE_OR_CONTROL), resp.text)).toEqual(2);
+
+    // The same NOC text is used for 'member of a firm' and 'control of firm' Nocs.
+    expect(stringCount(escapeNOCText(BO_NOC_MEMBER_OF_FIRM_OVER_25_PERCENT_OF_SHARES), resp.text)).toEqual(3);
+    expect(stringCount(escapeNOCText(BO_NOC_MEMBER_OF_FIRM_OVER_25_PERCENT_OF_VOTING_RIGHTS), resp.text)).toEqual(3);
+    expect(stringCount(escapeNOCText(BO_NOC_MEMBER_OF_FIRM_APPOINT_OR_REMOVE_MAJORITY_BOARD_DIRECTORS), resp.text)).toEqual(3);
+    expect(stringCount(escapeNOCText(BO_NOC_MEMBER_OF_FIRM_SIGNIFICANT_INFLUENCE_OR_CONTROL), resp.text)).toEqual(3);
+
+    // The same NOC text is used for 'person owner of land' and 'other entity owner of land' Nocs.
+    expect(stringCount(BO_NOC_JURISDICTION_ENGLAND_AND_WALES, resp.text)).toEqual(0);
+    expect(stringCount(BO_NOC_JURISDICTION_SCOTLAND, resp.text)).toEqual(0);
+    expect(stringCount(BO_NOC_JURISDICTION_NORTHERN_IRELAND, resp.text)).toEqual(0);
+  });
+
+  test.each([
+    [CHECK_YOUR_ANSWERS_URL],
+    [CHECK_YOUR_ANSWERS_WITH_PARAMS_URL]
+  ])(`renders the ${CHECK_YOUR_ANSWERS_PAGE} page with correct Beneficial Owner natures of control using url and the flag FEATURE_FLAG_ENABLE_PROPERTY_OR_LAND_OWNER_NOC on, %s`, async (url) => {
+    mockIsActiveFeature.mockReturnValue(true);
+
+    function escapeNOCText(input: string): string {
```

Comment:
I think we can extract this inner escapeNOCText function and move it to test/utils/test.utils.ts now there are 2 tests that use it.

Reply:
I can extract it but into this file because the two tests are in the same file, so there is no point in making it global

Reply:
Extracted
And fix the missing flag condition in the html table